src/group_theory/subgroup.lean

```diff
@@ -404,13 +406,10 @@ end
 @[to_additive] instance fintype_bot : fintype (⊥ : subgroup G) := ⟨{1},
 by {rintro ⟨x, ⟨hx⟩⟩, exact finset.mem_singleton_self _}⟩
 
-@[simp] lemma _root_.add_subgroup.card_bot : fintype.card (⊥ : add_subgroup A) = 1 :=
-fintype.card_eq_one_iff.2
-  ⟨⟨(0 : A), set.mem_singleton 0⟩, λ ⟨y, hy⟩, subtype.eq $ add_subgroup.mem_bot.1 hy⟩
-
 -- `@[to_additive]` doesn't work, because it converts the `1 : ℕ` to `0`.
-@[simp] lemma card_bot : fintype.card (⊥ : subgroup G) = 1 :=
-fintype.card_eq_one_iff.2
+@[to_additive, simp] lemma card_bot {h : fintype.{u} (⊥ : subgroup G)} :
+  @fintype.card (⊥ : subgroup G) h = 1 :=
+(@fintype.card_eq_one_iff (↥(⊥ : subgroup G)) h).2
```

Comment:
No need for the univeres annotations or `@`s:
```suggestion
@[simp] lemma _root_.subgroup.card_bot [fintype ↥(⊥ : subgroup A)] :
  fintype.card (⊥ : subgroup A) = 1 :=
fintype.card_eq_one_iff.2
```

Reply:
I swapped `simp` and `to_additive`. There is a need for the universe annotation, I don't understand why either. I did the curly brackets deliberately as sometimes you really don't want to use type class inference to infer that argument. For example, the following lemma only works with curly brackets

```lean
example (H : subgroup G) [h : fintype H] (hH : H = ⊥) : @fintype.card H h = 1 :=
begin
  simp only [hH],
  exact card_bot
end
```

Reply:
Thanks for the example, I'll try in gitpod tomorrow

Reply:
You're right, the `{}` seems to be needed for `exact` to work, but both versions work just fine with `rw`. A quick comment explaining the `{}` is fine if you want to keep them - but please remove the `@`s as I suggest in https://github.com/leanprover-community/mathlib/pull/8383#discussion_r674375932.